src/components/transition/Slider.tsx

```diff
@@ -0,0 +1,95 @@
+import { useCallback, useEffect, useRef, useState } from 'preact/hooks';
+
+import type { TransitionComponent } from '../../types';
+
+const Slider: TransitionComponent = ({
```

Comment:
I would like to be able to export this as `export default function Slider() {...}`, but then there's no way to type it as `TransitionComponent`.

It would be nice to be able to do something like `export default function Slider() {...} satisfies TransitionComponent` 😅